src/renderer/lib/downloadDolphin.ts

```diff
@@ -1,44 +1,84 @@
 import AdmZip from "adm-zip";
-import { spawn } from "child_process";
+import { spawn, spawnSync } from "child_process";
 import { download } from "common/download";
 import { fileExists } from "common/utils";
 import { remote } from "electron";
 import * as fs from "fs-extra";
 import path from "path";
+import { lt } from "semver";
 
-import { findDolphinExecutable, NETPLAY_PATH } from "./directories";
+import { DolphinType, findDolphinExecutable } from "./directories";
 
 export async function assertDolphinInstallation(
+  type: DolphinType,
   log: (message: string) => void
 ): Promise<void> {
   try {
-    await findDolphinExecutable();
-    log("Found existing Dolphin executable.");
+    await findDolphinExecutable(type);
+    log(`Found existing ${type} Dolphin executable.`);
+    log(`Checking if we need to update`);
+    const data = await getLatestReleaseData(type);
+    const latest_version = data.tag_name?.substring(1);
+    const is_outdated = await compareDolphinVersion(type, latest_version);
```

Comment:
You don't need to manually split out the `v` from the semver string. When using the `semver` package, it's smart enough to compare strings with or without the `v` prefix and the `compareDolphinVersion` calls `lt` from the `semver` package.

Also something something camelCase.

```suggestion
    const latestVersion = data.tag_name?.substring(1);
    const isOutdated = await compareDolphinVersion(type, latest_version);
```

Reply:
woah, that's nice